File: jetpack/src/main/java/com/pack/jetpack/MainActivity.java
package com.pack.jetpack;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import com.pack.jetpack.databinding.ActivityMainBinding;
import com.pack.jetpack.livedata.LiveDataActivity;
import com.pack.jetpack.navdemo.NavigationActivity;
import com.pack.jetpack.paging.PagingActivity;
import com.pack.jetpack.room.RoomActivity;
import com.pack.jetpack.viewmodeldemo.TestViewModelActivity;


public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        binding= DataBindingUtil.setContentView(this,R.layout.activity_main);
        myCLick();
    }

    private void myCLick() {
        binding.btViewModel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, TestViewModelActivity.class));
            }
        });
        binding.btLiveData.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, LiveDataActivity.class));
            }
        });
        binding.btNavigation.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, NavigationActivity.class));
            }
        });
        binding.btRoom.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, RoomActivity.class));
            }
        });
        binding.btLifecy.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, LifeActivity.class));
            }
        });
        binding.btPaging.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, PagingActivity.class));
            }
        });
    }
}

File: app/src/main/java/com/pack/safflower/MainActivity.kt
package com.pack.safflower

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.pack.safflower.base.BaseActivity_K
import com.pack.safflower.databinding.ActivityMainBinding

class MainActivity : BaseActivity_K() {
    lateinit var binding:ActivityMainBinding

    override fun setContentView() {
        binding=DataBindingUtil.setContentView(this,R.layout.activity_main)
    }

    override fun initData() {

    }

    override fun initClick() {
        binding.btSuc.setOnClickListener {
            binding.stateLayoutView
                    .Builder()
//                    .setLayout()
                    .hideAllState()
        }
        binding.btIng.setOnClickListener {
            binding.stateLayoutView
//                    获取Builder
                    .Builder()
//                    显示View
                    .showLoadIng()
//                    设置布局
//                    .setLayout(R.layout.layout)
//                    修改布局中的Img、提示语、按钮文字
                    .setIngImgOrTitleOrBt(R.mipmap.ic_launcher,"111","222")
        }
        binding.btFail.setOnClickListener {
            binding.stateLayoutView
                    .Builder()
//                    .setLayout()
                    .showLoadFail()
        }
        binding.btError.setOnClickListener {
            binding.stateLayoutView
                    .Builder()
//                    .setLayout()
                    .showLoadError()
        }
        binding.btEmpty.setOnClickListener {
            binding.stateLayoutView
                    .Builder()
//                    .setLayout()
                    .showLoadEmpty()
        }
    }
}